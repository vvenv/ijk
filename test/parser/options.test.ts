import { describe, expect, test } from 'vitest';
import { parse } from './__helper';

describe('trimWhitespace', () => {
  test('on', () => {
    expect(
      parse(
        `{{ for name in names }}
  {{ name }} in {{ names }}
{{ endfor }}`,
      ),
    ).toMatchSnapshot();
  });

  test('off', () => {
    expect(
      parse(
        ` {{ for name  in names }}
  {{ name }} in  {{ names }}
 {{ endfor  }} `,
        {
          trimWhitespace: false,
        },
      ),
    ).toMatchSnapshot();
  });
});

test('tagStart and tagEnd', () => {
  expect(
    parse(`{% for name in names %}{{ name }} in {{ names }}{% endfor %}`, {
      tagStart: '{%',
      tagEnd: '%}',
    }),
  ).toMatchSnapshot();
});
