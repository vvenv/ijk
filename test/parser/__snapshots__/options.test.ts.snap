// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tagStart and tagEnd 1`] = `""use strict";let s = "";const k_1_0=Object.keys(c.names);const l_1_0=k_1_0.length;for (let i_1_0=0; i_1_0<l_1_0; i_1_0++) {const c_i_1_0={...c,name: c.names[k_1_0[i_1_0]],loop: {index: i_1_0,first: i_1_0 === 0,last: i_1_0 === l_1_0,length: l_1_0}};s+="{{ name }} in {{ names }}";}return s;"`;

exports[`trimWhitespace > off 1`] = `""use strict";let s = "";s+=" ";const k_1_0=Object.keys(c.names);const l_1_0=k_1_0.length;for (let i_1_0=0; i_1_0<l_1_0; i_1_0++) {const c_i_1_0={...c,name: c.names[k_1_0[i_1_0]],loop: {index: i_1_0,first: i_1_0 === 0,last: i_1_0 === l_1_0,length: l_1_0}};s+="\\n  ";s+=e(c_i_1_0.name);s+=" in  ";s+=e(c_i_1_0.names);s+="\\n ";}s+=" ";return s;"`;

exports[`trimWhitespace > on 1`] = `""use strict";let s = "";const k_1_0=Object.keys(c.names);const l_1_0=k_1_0.length;for (let i_1_0=0; i_1_0<l_1_0; i_1_0++) {const c_i_1_0={...c,name: c.names[k_1_0[i_1_0]],loop: {index: i_1_0,first: i_1_0 === 0,last: i_1_0 === l_1_0,length: l_1_0}};s+=" ";s+=e(c_i_1_0.name);s+=" in ";s+=e(c_i_1_0.names);s+=" ";}return s;"`;
